udata/core/user/api.py

```diff
@@ -226,51 +227,63 @@ def get(self):
         search_parser.parse_args()
         return search.query(UserSearch, **multi_to_dict(request.args))
 
-    @api.secure
+    @api.secure(admin_permission)
     @api.doc('create_user')
     @api.expect(user_fields)
     @api.marshal_with(user_fields, code=201)
     @api.response(400, 'Validation error')
     def post(self):
         '''Create a new object'''
-        form = api.validate(UserProfileForm)
+        form = api.validate(UserProfileAdminForm)
         user = form.save()
         return user, 201
 
 
 @ns.route('/<user:user>/', endpoint='user')
 @api.response(404, 'User not found')
-@api.response(410, 'User has been deleted')
+@api.response(410, 'User is not active or has been deleted')
 class UserAPI(API):
+
+    def check_permissions(self, user, readonly=False):
+        if not user.visible:
+            # We cannot modify or read an inactive user, unless we are admin
+            api.abort(410, 'User is not active')
+        elif user.deleted and (
+            # We cannot modify a deleted user
+            not readonly or (
+                # We cannot get a deleted user if we do not have the
+                # permissions
+                readonly and not UserEditPermission(user).can())
+        ):
+            api.abort(410, 'User has been deleted')
+        elif not readonly:
+            # We must have the permission to modify the user
+            UserEditPermission(user).test()
```

Comment:
I'm always experimenting on that kind of `elif`-based permissions, here is a proposition:

```python
def check_permissions(self, user, readonly=False):
    user_edit_permission = UserEditPermission(user)
    is_active = not user.visible
    is_editable = not readonly
    admin_access = readonly and not user_edit_permission.can()
    has_been_deleted = user.deleted and (is_editable or admin_access)
    if is_active:
        api.abort(410, 'User is not active')
    elif has_been_deleted:
        api.abort(410, 'User has been deleted')
    elif is_editable:
        user_edit_permission.test()
```
The goal is to clarify what is done in the second part without having too much imbricated tests that are blurring the algorithm. Let's discuss this, as I said I'm not sure that it's more readable at the end.